File: example/scalalib/testing/1-test-suite/foo/test/src/FooMoreTests.scala
package foo
import utest._
object FooMoreTests extends TestSuite {
  def tests = Tests {
    test("hello") {
      val result = Foo.hello()
      assert(result.startsWith("Hello"))
      result
    }
    test("world") {
      val result = Foo.hello()
      assert(result.endsWith("World"))
      result
    }
  }
}

File: example/kotlinlib/testing/2-test-suite/foo/test/src/foo/FooMoreTests.kt
package foo

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import io.kotest.matchers.string.shouldEndWith
import io.kotest.matchers.string.shouldStartWith
import org.mockito.kotlin.doReturn
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify
import org.mockito.kotlin.whenever

class FooMoreTests :
    FunSpec({

        test("hello") {
            val result = Foo().hello()
            result shouldStartWith "Hello"
        }

        test("world") {
            val result = Foo().hello()
            result shouldEndWith "World"
        }

        test("mockito") {
            val mockFoo = mock<Foo>()

            whenever(mockFoo.hello()) doReturn "Hello Mockito World"

            val result = mockFoo.hello()

            result shouldBe "Hello Mockito World"
            verify(mockFoo).hello()
        }
    })
